src/Wordpress/CustomPostType/Booking.php

```diff
@@ -167,130 +180,178 @@ public function saveAdminBookingFields( $post_id, $post = null, $update = null )
-					\CommonsBooking\Model\Timeframe::META_LOCATION_ID => $locationId,
-					\CommonsBooking\Model\Timeframe::META_ITEM_ID => $itemId,
-					\CommonsBooking\Model\Timeframe::REPETITION_START => $startDate,
-					\CommonsBooking\Model\Timeframe::REPETITION_END => $endDate,
-					'type' => Timeframe::BOOKING_ID,
-                ];
+		// add internal comment if admin edited booking via frontend
+		if ( $booking && $booking->post_author !== '' && intval( $booking->post_author ) !== intval( get_current_user_id() ) ) {
+			$postarr['meta_input']['admin_booking_id'] = get_current_user_id();
+			$internal_comment                          = esc_html__( 'status changed by admin user via frontend. New status: ', 'commonsbooking' ) . $post_status;
+			$booking->appendToInternalComment( $internal_comment, get_current_user_id() );
+		}
 
-                $postId = wp_insert_post( $postarr, true );
+		$postarr['type']                  = $postType;
+		$postarr['post_status']           = $post_status;
+		$postarr['post_type']             = self::getPostType();
+		$postarr['post_title']            = esc_html__( 'Booking', 'commonsbooking' );
+		$postarr['meta_input']['comment'] = $comment;
+
+		// New booking
+		if ( empty( $booking ) ) {
+			$postarr['post_name']  = Helper::generateRandomString();
+			$postarr['meta_input'] = array(
+				\CommonsBooking\Model\Timeframe::META_LOCATION_ID => $locationId,
+				\CommonsBooking\Model\Timeframe::META_ITEM_ID     => $itemId,
+				\CommonsBooking\Model\Timeframe::REPETITION_START => $repetitionStart,
+				\CommonsBooking\Model\Timeframe::REPETITION_END   => $repetitionEnd,
+				'type'                                            => Timeframe::BOOKING_ID,
+			);
 
-                // Existing booking
-            } else {
-                $postarr['ID'] = $booking->ID;
-                $postId        = wp_update_post( $postarr );
+			$postId = wp_insert_post( $postarr, true );
 
-            }
+			// Existing booking
+		} else {
+			$postarr['ID'] = $booking->ID;
+			$postId        = wp_update_post( $postarr );
 
-            $this->saveGridSizes( $postId, $locationId, $itemId, $startDate, $endDate );
+		}
 
-            $bookingModel = new \CommonsBooking\Model\Booking( $postId );
-            // we need some meta-fields from bookable-timeframe, so we assign them here to the booking-timeframe
-            $bookingModel->assignBookableTimeframeFields();
+		self::saveGridSizes( $postId, $locationId, $itemId, $repetitionStart, $repetitionEnd );
 
-            // get slug as parameter
-            $post_slug = get_post( $postId )->post_name;
+		$bookingModel = new \CommonsBooking\Model\Booking( $postId );
+		// we need some meta-fields from bookable-timeframe, so we assign them here to the booking-timeframe
+		try {
+			$bookingModel->assignBookableTimeframeFields();
+		} catch ( \Exception $e ) {
+			throw new BookingDeniedException( __( 'There was an error while saving the booking. Please try again.', 'commonsbooking' ) );
```

Comment:
Ehe eine Frage als eine Anmerkung: Gehen die exakten Exceptions hier nicht verloren? Also für den technischen Support danach. Bzw. wir sehen zwar die Zeile dieser Exception, aber halt nicht welche in z.B. `assignBookableTimeframeFields` geworfen wurde.

Reply:
Das stimmt, ich ergänze mal den geworfenen Fehler.